rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,154 @@
+This CND can be used to add perfectly calibrated noise to a statistic to achieve $f$-DP~\parencite[Proposition~2.2]{dong2019gaussian}. 
+
+\begin{thm} \label{thm1}  % Thm 3.9
+    \cite[Theorem~3.9]{awan2023canonical} \\
+    Let $f$ be a symmetric nontrivial tradeoff function and let $F_f$ be as in Definition~\ref{def1}~\parencite{awan2023canonical}. 
+    Then, as stated in \parencite[Theorem~3.9]{awan2023canonical}, $F_f$ is a canonical noise distribution for $f$. 
+\end{thm}
+
+In Definition~\ref{def1}~\parencite{awan2023canonical}, we explained the cdf of the CND we made. 
+This explanation helps us understand the distribution's features, but when it comes to sampling, the quantile function is key. 
+In Proposition~\ref{prop1}~\parencite{awan2023canonical} provides a recursive formula for the CND's quantile function,
+as described in Definition~\ref{def2}~\parencite{awan2023canonical}, and show that we can finish it in just a few steps.
+
+\begin{prop}\label{prop1}   % Prop F.6
+    \cite[Proposition~F.6]{awan2023canonical} \\
+    Let $f$ be a symmetric nontrivial tradeoff function and let $F_f$ be as in Definition~\ref{def1}. 
+    Then the quantile function $F_f^{-1}:(0,1)\rightarrow \mathbb{R}$ for $F_f$ can be expressed as
+    \[F_f^{-1}(u) = \begin{cases}
+    F_f^{-1}(1-f(u))-1&u<c\\
+    \frac{u-1/2}{1-2c}&c\leq u\leq 1-c\\
+    F_f^{-1}(f(1-u))+1&u>1-c,
+    \end{cases}\]
+    where $c$ is the unique fixed point of $f$. 
+    {Furthermore, for any $u\in (0,1)$, the expression $Q_f(u)$ takes a finite number of recursive steps to evaluate. Thus,} 
+    if $U\sim U(0,1)$, then $F_f^{-1}(U) \sim F_f$. 
+\end{prop}
+
+% Cor 3.10
+According to Corollary~3.10 in \cite{awan2023canonical}, 
+the distribution $\mathrm{Tulap}(0,b,q)$, where $b=\exp(-\epsilon)$ and $q = \frac{2\delta b}{1-b+2\delta b}$ 
+is a CND for $f_{\epsilon,\delta}$-DP, which agrees with the construction of Definition~\ref{def1}~\parencite{awan2023canonical}. 
+
+From the definition, it is easy to verify that the cdf of a Tulap random variable agrees with $F_f$ on $[-1/2,1/2]$. 
+Tulap cdf also satisfies the recurrence relation of Definition~\ref{def1}~\parencite{awan2023canonical}. 
+Note that the cdf of $\mathrm{Tulap}(0,b,0)$ is 
+\[F_{N_0}(x) = \begin{cases}
+    \frac{b^{-[x]}}{1+b}(b+\{x-[x]+1/2\}(1-b))& x\leq 0\\
+    1- \frac{b^{[x]}}{1+b}(b+\{[x]-x+1/2\}(1-b))&x>0,
+\end{cases}\]
+where $[x]$ is the nearest integer function. 
+The cdf of $\mathrm{Tulap}(0,b,q)$ is
+\[F_N(x) = \begin{cases}
+    0&F_{N_0}(x)<q/2\\
+    \frac{F_{N_0}(x)-q/2}{1-q}& q/2\leq F_{N_0}(x)\leq 1-q/2\\
+    1&F_{N_0}(x)>1-q/2.
+\end{cases}\]
+By inspection, the fixed point of $f_{\epsilon,\delta}$ is $c=\frac{1-\delta}{1+e^\epsilon}$. 
+It is easy to verify that $F_N(x) = c(1/2-x) + (1-c)(x+1/2)$ for $x\in (-1/2,1/2)$. 
+We then have that $F_N$ satisfies the recurrence relation in Definition~\ref{def1}~\parencite{awan2023canonical}. 
```

Comment:
No need to cite the paper again. Hurts the readability and is distracting.

Reply:
Gotta get those cites up!